ONSdigital/takeon-business-layer
# Feature/spp 828 dateadjustment

# Description
This PR covers the business layer changes for the Date Adjustment processing. This includes the two API calls one for getting configuration, another for saving adjusted responses and corresponding unit tests.
Story Link: https://collaborate2.ons.gov.uk/jira/browse/SPP-828

# How to test?
Before executing Lambda
1. Go to UI 
2. Go to RSI survey and any form for example Form 6 and Period 201903 - Any reference is OK
3. Q11 and Q12 values are mandatory and for period 201903 enter 
    Period start date = 20190304
    Period end date = 20190328
4. For period 201904 period enter 
Period start date = 20190404
Period end date  = 20190428
5. And also enter the values for the numerical questions Q20 to Q26
6. Save the Form
7. To test this, you can use the SPP-828 environment and execute the Lambda **'spp-es-takeon-validation-spp828-date-adjustment'** manually by updating the test event with the following JSON Payload and checking the Cloudwatch logs:
  
{ "reference": "11111111111", "period": "201904", "survey": "023" }

*** Ask me for references that work / loaded into the database.
8. Refresh the UI for the above reference period and Form and you can see the AdjustedResponse if any
9. Verify database 
  Response Table - Column AdjustedResponse and AverageWeeklyAdjustedResponse
  ContributorDateAdjustment Table - The corresponding score flags and audit information.

# Unit Tests
You can run java unit tests using:   mvn clean test

# Code Coverage
You can verify code coverage using:   awk -F"," '{ instructions += $4 + $5; covered += $5 } END { print covered, "/", instructions, " instructions covered"; print 100*covered/instructions, "% covered" }' target/site/jacoco/jacoco.csv

# Check Style errors
You can verify check style errors using :   mvn package install

# Links
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?pageId=17467155
